### PR TITLE
update phoenix and other apps so easier rendering of nested live_components is possible

### PR DESCRIPTION
**Pull request by bbrewer-papa-x**
smartcitiesdata/smartcitiesdata/issues/577

### CODE REVIEW

File: apps/andi/lib/andi_web/live/edit_live_view.ex [hunk 1]
```diff
@@ -191,7 +201,7 @@ defmodule AndiWeb.EditLiveView do
     |> complete_validation(socket)
   end
 
-  def handle_event("validate", %{"form_data" => form_data}, socket) do
+  def handle_event("validate", %{"form_data" => form_data} = stuff, socket) do
```
Comment (mcsearchin) [inline apps/andi/lib/andi_web/live/edit_live_view.ex]: remove

File: apps/andi/lib/andi_web/router.ex [hunk 2]
```diff
@@ -12,7 +12,7 @@ defmodule AndiWeb.Router do
     plug :accepts, ["html"]
     plug :fetch_session
     plug :fetch_flash
-    plug Phoenix.LiveView.Flash
+    # plug Phoenix.LiveView.Flash
```
Comment (mcsearchin) [inline apps/andi/lib/andi_web/router.ex]: Should this remain commented out?

File: apps/andi/test/unit/andi_web/live/edit_live_view_test.exs [hunk 3]
```diff
@@ -785,26 +784,46 @@ defmodule AndiWeb.EditLiveViewTest do
...
 
     {value, text}
   end
 
+  defp find_element(html, selector) do
+    html
+    |> Floki.parse_fragment!()
+    |> Floki.find(selector)
+  end
+
```
Comment (mcsearchin) [inline apps/andi/test/unit/andi_web/live/edit_live_view_test.exs]: Thanks for all of these helpers.

File: nested.html [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+          <div class='expander cell'>-------+</div>
+          <div class='content cell'>Nested Row 2</div>
+        </div>
+        
+      </div>
+    </div>
+  </div>
+</body>
+
+</html>
```
Comment (mcsearchin) [inline nested.html]: Should this be included in this PR?

File: apps/andi/lib/andi/input_schemas/dataset_input.ex [hunk 5]
```diff
@@ -47,7 +48,9 @@ defmodule Andi.InputSchemas.DatasetInput do
 
   @key_value_type_keys [:sourceQueryParams, :sourceHeaders]
 
-  @non_embedded_types Map.drop(@types, @key_value_type_keys)
+  @embedded_types @key_value_type_keys
```
Comment (mcsearchin) [inline apps/andi/lib/andi/input_schemas/dataset_input.ex]: Assuming that the idea is that this will eventually include the data dictionary type as well?

**Comment by bbrewer-papa-x** [comment]
Thanks @mcsearchin - yeah I'll remove most of the stuff that is weird, like `nested.html`. I got some cleanin' to do on this

File: apps/andi/lib/andi_web/router.ex [hunk 6]
```diff
@@ -26,7 +25,7 @@ defmodule AndiWeb.Router do
     pipe_through :browser
 
     get "/", Redirect, to: "/datasets"
-    live "/datasets", DatasetLiveView, session: [:path_params]
+    live "/datasets", DatasetLiveView, layout: {AndiWeb.LayoutView, :app}
```
Comment (dunson062786) [inline apps/andi/lib/andi_web/router.ex]: What does this do?

Comment (bbrewer-papa-x) [reply apps/andi/lib/andi_web/router.ex]: With the new live view version, if that is not set, then the dataset list page will not include the default andi html layout that includes styling and other things.

Comment (dunson062786) [reply apps/andi/lib/andi_web/router.ex]: makes sense

**Pull request merged**